document/src/main/java/com/yahoo/vespaxmlparser/FeedOperation.java

```diff
@@ -1,17 +1,16 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
 package com.yahoo.vespaxmlparser;
 
 import com.yahoo.document.Document;
 import com.yahoo.document.DocumentId;
 import com.yahoo.document.DocumentUpdate;
 import com.yahoo.document.TestAndSetCondition;
 
-public abstract class FeedOperation {
+public class FeedOperation {
```

Comment:
How about keeping this and the getters below abstract and create an InvalidFeedOperation subclass instead?

Reply:
The reason it is not abstract is that the implementations should only implement the one they actually support, while the other will return null.
It would be strange for them to extend Invalid.....
Perhaps if I named it DefaultFeedOperation and made it invalid.

Reply:
Anyway I will merge it now, and I can have another iteration tomorrow.
It is not a public API anyway.